server/src/main/kotlin/com/mobilejazz/colloc/domain/interactor/CollocInteractor.kt

```diff
@@ -43,25 +41,20 @@ class CollocInteractor(
     return zip
   }
 
-  private suspend fun downloadCsv(link: URL): File = downloadFileInteractor(link, "downloaded_google_drive_file.csv")
-
-  private fun generateAngularLocales(csvFile: File, output: File) {
-    val parse = ParseCsvInteractor()
-    val result = parse(csvFile.absolutePath)
-    val json = Json { }
-    val angularFolder = File(output.absolutePath + "/angular/")
-    angularFolder.mkdirs()
-    for (content in result) {
-      val serializer = MapSerializer(String.serializer(), String.serializer())
-      val languageJson = json.encodeToString(serializer, content.value)
-      val languageFile = File("${angularFolder.absolutePath}/${content.key}.json")
-      languageFile.writeText(languageJson)
-    }
+  private suspend fun Platform.generateLocales(id: String, outputDirectory: File) {
+    val csvFile = downloadCsv(id.toFullImportedFileURL())
+    val dictionary = localizationDecoder.decode(csvFile.absolutePath)
+    platformEncodeInteractorMap.getValue(this).invoke(outputDirectory, dictionary)
   }
 
+  private suspend fun downloadCsv(link: URL): File = downloadFileInteractor(
+    link,
+    File("downloaded_google_drive_file.csv.tsv").createFile()
```

Comment:
`.csv.tsv` why is needed?

Reply:
I just copy pasted it from `DownloadFileInteractor` to here, will check if it works without any of the extensions

Reply:
OK, resolve conversation with the outcome

Reply:
Check, removed `.csv` part